src/components/v5/common/ActionSidebar/partials/forms/EditColonyDetailsForm/hooks.ts

```diff
@@ -23,26 +30,63 @@ export const useEditColonyDetails = (
     name: DECISION_METHOD_FIELD_NAME,
   });
 
+  const defaultValues = useMemo<DeepPartial<EditColonyDetailsFormValues>>(
+    () => ({
+      colonyName: metadata?.displayName,
+      avatar: {
+        image: metadata?.avatar,
+        thumbnail: metadata?.thumbnail,
+      },
+      colonyDescription: metadata?.description || '',
+      createdIn: Id.RootDomain.toString(),
+      externalLinks: metadata?.externalLinks ?? [],
+    }),
+    [
+      metadata?.avatar,
+      metadata?.description,
+      metadata?.displayName,
+      metadata?.externalLinks,
+      metadata?.thumbnail,
+    ],
+  );
+
+  const validationSchema = useMemo(
+    () =>
+      object()
+        .shape({
+          avatar: object().nullable().shape({
+            image: string().nullable().defined(),
+            thumbnail: string().nullable().defined(),
+          }),
+          colonyName: string().trim().max(MAX_COLONY_DISPLAY_NAME).required(),
+          createdIn: string().defined(),
+          decisionMethod: string().defined(),
+          description: string().max(MAX_ANNOTATION_LENGTH).notRequired(),
+          colonyDescription: string().required(),
+          externalLinks: array()
+            .of(
+              object()
+                .shape({
+                  name: string().defined().oneOf(Object.values(ExternalLinks)),
+                  link: string().url().required(),
+                })
+                .defined(),
+            )
+            .required(),
+        })
+        .hasValuesChanged(
```

Comment:
why do we actually need this? isn't the `touched` or `isDirty` form state enough?

Reply:
@jakubcolony  in comments above said that we should `aim to put all validation in validation schema`. With `isDirty`, the user could enter something and delete it and the validation would pass.